### PR TITLE
chore: provide new sub value for V2 domain 

### PR DESCRIPTION
**Pull request by leleueri**
This PR change the way the sub value is generated for token delivered by domain in version 2
This change is a first step to help us to manage better resiliency on AM GW.

fixes AM-3472

### CODE REVIEW

File: gravitee-am-extensiongrant/gravitee-am-extensiongrant-jwtbearer/src/main/java/io/gravitee/am/extensiongrant/jwtbearer/provider/JWTBearerExtensionGrantProvider.java
[the remark this answers is not in the document]
Comment (leleueri) [reply gravitee-am-extensiongrant/gravitee-am-extensiongrant-jwtbearer/src/main/java/io/gravitee/am/extensiongrant/jwtbearer/provider/JWTBearerExtensionGrantProvider.java:1]: manage to create a dedicated implementation of ExtensionGrantGranter for DomainV2.

File: gravitee-am-gateway/gravitee-am-gateway-handler/gravitee-am-gateway-handler-common/src/main/java/io/gravitee/am/gateway/handler/common/vertx/web/auth/handler/impl/OAuth2AuthHandlerImpl.java [hunk 1]
```diff
@@ -97,36 +107,52 @@ public void handle(RoutingContext context) {
...
-                if (selfResource) {
-                    final String resourceId = context.request().getParam(resourceParameter);
-                    boolean isResourceIdValid = resourceId != null && resourceId.equals(token.getSub());
-                    boolean hasRequiredScope = resourceRequiredScope == null || token.hasScope(resourceRequiredScope);
+                var single = Single.just(Optional.of(token.getSub()));
+                if (selfResource && isSelfResourceAUser) {
+                    single = this.subjectManager.findUserIdBySub(token.getSub())
+                            .map(Optional::ofNullable)
+                            .switchIfEmpty(single);
+                }
```
Comment (lgw-gravitee) [inline gravitee-am-gateway/gravitee-am-gateway-handler/gravitee-am-gateway-handler-common/src/main/java/io/gravitee/am/gateway/handler/common/vertx/web/auth/handler/impl/OAuth2AuthHandlerImpl.java:116]: Maybe<Optional<?>> can be simplified
`                Maybe<String> single = Maybe.just(token.getSub());
                if (selfResource && isSelfResourceAUser) {
                    single = this.subjectManager.findUserIdBySub(token.getSub())
                            .switchIfEmpty(single);
                }`

**Review by lgw-gravitee — APPROVED** [review]

File: gravitee-am-gateway/gravitee-am-gateway-handler/gravitee-am-gateway-handler-users/src/main/java/io/gravitee/am/gateway/handler/users/resources/consents/UserConsentEndpointHandler.java [hunk 2]
```diff
@@ -52,11 +57,18 @@ public void get(RoutingContext context) {
...
      */
     public void revoke(RoutingContext context) {
+        final JWT accessToken = context.get(ConstantKeys.TOKEN_CONTEXT_KEY);
         final String userId = context.request().getParam("userId");
         final String consentId = context.request().getParam("consentId");
 
+        final var singleUserId = getUserIdFromSub(accessToken);
         getPrincipal(context)
-                .flatMapCompletable(principal -> userService.revokeConsent(userId, consentId, principal))
+                .flatMapCompletable(principal -> singleUserId.flatMapCompletable(id -> {
```
Comment (podlesrafal) [inline gravitee-am-gateway/gravitee-am-gateway-handler/gravitee-am-gateway-handler-users/src/main/java/io/gravitee/am/gateway/handler/users/resources/consents/UserConsentEndpointHandler.java:66]: What about inline `singleUserId` to getPrincipal stream, like this? 
```suggestion
                .flatMapCompletable(principal -> getUserIdFromSub(accessToken).flatMapCompletable(id -> {
```

**Review by podlesrafal — APPROVED** [review]
Overall code looks ok. Just small remarks.

**Pull request merged**